File: 프로그래머스/lv1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.js
function reArrange(n){
    // 이 모든 과정을 한 줄로 정리 가능
     return Number(n.toString().split('').sort().reverse().join(''))
    
//     // 정수를 문자열로 변환
//     var newString = n.toString();
//     console.log(newString);
//     // 문자열을 배열로 변환
//     var reArray = newString.split('');
//     // 배열을 크기 순서대로 재정렬
//     reArray.sort();
//     // 오름차순을 reverse하기
//     reArray.reverse();
//     // 배열을 문자열로
//     newString = reArray.join('');
//     // 문자열을 정수로
//     return Number(newString);
}

function solution(n) {
    return reArrange(n);
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.py
def solution(n):
    answer = 0
    
    list_n = list(str(n))  # 문자열 변환 후 list 에 저장 # ['1', '1', '8', '3', '7', '2']
    list_n.sort(reverse=True) # 내림차순으로 정렬
    list_str = ''.join(list_n) # 문자열로 결합
    answer = int(list_str) # 정수로 변환
    
    return answer
